elixir/apps/web/lib/web/live/settings/identity_providers/openid_connect/components.ex

```diff
@@ -95,12 +100,47 @@ defmodule Web.Settings.IdentityProviders.OpenIDConnect.Components do
                 </div>
               </.inputs_for>
             </div>
-
-            <.submit_button>
-              Connect Identity Provider
-            </.submit_button>
           </:content>
         </.step>
+        <.step :if={@show_sync_msg}>
+          <:title>
+            Step 3. Enable Directory Sync <.icon name="hero-arrow-path" class="w-5 h-5 ml-2" />
+            <span class="inline-flex w-1/2 justify-end">
+              <.link
+                navigate={~p"/#{@account}/settings/billing"}
+                class="text-sm text-primary-500 ml-2"
+              >
+                <.badge type="primary" title="Feature available on a higher pricing plan">
+                  <.icon name="hero-lock-closed" class="w-3.5 h-3.5 mr-1" /> UPGRADE TO UNLOCK
+                </.badge>
+              </.link>
+            </span>
+          </:title>
+          <:content>
+            Directory sync is not enabled on your current plan.
+            <div class="blur-md">
```

Comment:
nit: current blur amount seems a little jarring (maybe it's just me?) 😄 

```suggestion
            <div class="blur-sm">
```

Reply:
👍 Updated.

<img width="735" alt="Screenshot 2024-05-14 at 12 21 07 PM" src="https://github.com/firezone/firezone/assets/2646332/8e167737-edbd-46f1-9f6b-70d406245670">